pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/PredicateUtils.java

```diff
@@ -76,7 +76,7 @@ public static IntSet getDictIdSet(BaseInPredicate inPredicate, Dictionary dictio
     IntSet dictIdSet = new IntOpenHashSet(hashSetSize);
     switch (dataType.getStoredType()) {
       case INT:
-        int[] intValues = inPredicate.getIntValues();
+        int[] intValues = dataType == DataType.BOOLEAN ? inPredicate.getBooleanValues() : inPredicate.getIntValues();
```

Comment:
Good catch! The clean fix should be change the switch to switch on `dataType` so that we can handle `BOOLEAN` and `TIMESTAMP` properly.
Do you need this change for the purpose of this PR? If not, I'd suggest having a separate PR just for this fix so that the scope is clear

Reply:
@Jackie-Jiang I'd rather fix it here since we use membership operators `IN (value, value)` in many of our queries.